compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -1789,6 +1825,9 @@ fn expr_resolve(
                 let (expr_id, _) = elaborator.elaborate_expression(expr);
                 Value::TypedExpr(TypedExpr::ExprId(expr_id))
             }
+            ExprValue::Pattern(_) => {
+                todo!("Can't resolve a pattern");
```

Comment:
Should we give a compile-error here? What should it say? Or maybe `Pattern` should be a separate type?

Reply:
If we can't use them in other places where we expect to use expressions then I suppose we can just make them a separate type.

It's a bit unfortunate since I think the only variant that can't be represented as an expression would be the `mut` pattern. We could potentially have a `try_as_expression` I suppose then error if it can't be converted.